pkg/util/cli/deprecated.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package cli
+
+import (
+	"fmt"
+	"io"
+	"strings"
+)
+
+func deprecatedFlagWarning(deprecated, replaceWith string) string {
+	return fmt.Sprintf("WARNING: `%s` argument is deprecated and will be removed in a future version. Please use `%s` instead.\n", deprecated, replaceWith)
+}
+
+type ReplaceFlag struct {
+	Hint string
+	Args []string
+}
+
+type ReplaceFlagFunc func(arg string, flag string) ReplaceFlag
```

Comment:
It's a very small detail but inverting the two argument would make more sens I think.

We do `--flag=value` not `value=--flag`.

Reply:
It's confusing all around because you're replacing one flag with another flag, so I feel like whatever combination we end up using will probably be a little confusing. In any case, I didn't want to touch any of this because I wanted to keep the refactored code as-is for smooth review. I'm all up for fixing all of your feedback in a follow-up PR I could put together immediately.